pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -2795,6 +2802,111 @@ protected void internalSetNamespaceResourceGroup(String rgName) {
+        AtomicInteger totalCount = new AtomicInteger();
+        AtomicInteger totalErrors = new AtomicInteger();
+        AtomicInteger totalUnknown = new AtomicInteger();
+        managedLedgerOffloader.scanLedgers((md -> {
+            log.info("Found ledger {}", md);
+            Map<String, Object> objectInfo = new HashMap<>();
+            objectInfo.put("ledger", md.getLedgerId());
+            objectInfo.put("name", md.getName());
+            objectInfo.put("uri", md.getUri());
+            objectInfo.put("uuid", md.getUuid());
+            objectInfo.put("size", md.getSize());
+            objectInfo.put("lastModified", md.getLastModified());
+            objectInfo.put("userMetadata", md.getUserMetadata());
+
+            String status = "UNKNOWN";
+
+            if (md.getUserMetadata() != null) {
+                // non case sensistive
+                TreeMap<String, String> userMetadata = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+                userMetadata.putAll(md.getUserMetadata());
+                String clusterName = userMetadata.get(LedgerOffloader.METADATA_PULSAR_CLUSTER_NAME);
+                if (localClusterName.equals(clusterName)) {
+                    String managedLedgerName = userMetadata.get("managedledgername");
+                    if (managedLedgerName != null) {
+                        objectInfo.put("managedLedgerName", managedLedgerName);
+                        try {
+                            status = checkLedgerShouldBeOnTieredStorage(md.getLedgerId(), md.getUuid(),
+                                    managedLedgerName, objectInfo, pulsar().getManagedLedgerFactory());
+                        } catch (InterruptedException err) {
+                            Thread.currentThread().interrupt();
+                            throw new RuntimeException(err);
+                        } catch (ManagedLedgerException err) {
+                            log.error("Error while checking managed ledger {}", managedLedgerName);
+                            throw new RuntimeException(err);
+                        }
+                    }
+                }
+            }
+            totalCount.incrementAndGet();
+            objectInfo.put("status", status);
+            switch (status) {
+                case "OK":
+                    break;
+                case "UNKNOWN":
+                    totalUnknown.incrementAndGet();
+                    break;
+                default:
+                    totalErrors.incrementAndGet();
+                    break;
+            }
+
+            objects.add(objectInfo);
+            return true;
+        }), managedLedgerOffloader.getOffloadDriverMetadata());
+        topLevelResult.put("errors", totalErrors.intValue());
+        topLevelResult.put("total", totalCount.intValue());
+        topLevelResult.put("unknownObjects", totalUnknown.intValue());
+        log.info("internalScanOffloadedLedgers {} scan finished");
+
+        return topLevelResult;
```

Comment:
The `topLevelResult` holds too much infos, and return to the client by REST is not a good solution. The number of items in `topLevelResult` depends on how many topics in this namespace.

Reply:
see my comment here https://github.com/apache/pulsar/pull/14930#discussion_r839373087

I can convert to streaming style in this patch if you prefer

Reply:
done